Add the Request Details card markup

rememberlenny opened the pull request:
# Important

Requires merging of other Details Page PR's before

## Content

Contains partials related to Request Details card

## Trello Reference Link
https://trello.com/c/0OrpiMHD 

- Request Details Card Markup

jessieay commented on app/views/proposals/detail/_request_details.html.haml:
```diff
@@ -0,0 +1,85 @@
+.row
+  .medium-12.column
+    .card#request-details-card
+      - if @client_data_instance && @proposal.client_slug == "ncr"
```
we only want this to work for NCR for now?

pkarman replied:
piling on: if we are hardcoding client-specific values in the templates, we are doing it wrong.

rememberlenny replied:
@jessieay @pkarman Not sure how that originated, but is not needed. Fixed!

jessieay commented on app/views/proposals/detail/_request_details.html.haml:
```diff
@@ -0,0 +1,85 @@
+                .detail-display
+                  %p.request-detail.detail-value{id: 'description-' + @client_data_instance.id.to_s }
+                    = @client_data_instance.description
+
```
extra newline here

rememberlenny replied:
@jessieay Fixed!

jessieay commented on app/views/proposals/detail/_request_details.html.haml:
```diff
@@ -0,0 +1,85 @@
+                      .detail-display
+                        %span.detail-value{id: 'cl_number-' + @client_data_instance.id.to_s }
+                          = @client_data_instance.cl_number || "-"
+
```
extra newline

rememberlenny replied:
@jessieay Plop

jessieay commented on app/views/proposals/detail/_request_details.html.haml:
```diff
@@ -0,0 +1,85 @@
+                          input_html: { class: "js-selectize required", data: { initial: JSON.generate(vendor_options(@client_data_instance.vendor)) }}
+                      .detail-display
+                        %span.detail-value{id: 'vendor-' + @client_data_instance.id.to_s }
+                          = @client_data_instance.vendor || "-"
```
we have a method in `app/helpers/mailer_helper.rb` (`property_display_value`) that takes care of this "value or dash" logic. perhaps we can move that to a different helper and use it here? (could also use method from mailer helper here but that seems confusing).

jessieay replied:
same for cl_number below

jessieay commented on app/views/proposals/detail/_request_details.html.haml:
```diff
@@ -0,0 +1,85 @@
+                    .detail-wrapper
+                      %label.detail-element Building number
+                      .detail-edit
+                        = f.input :building_number,
```
looks like this is in `app/views/proposals/detail/fields/_building_number.html.haml` already?

jessieay commented on app/views/proposals/detail/_request_details.html.haml:
```diff
@@ -0,0 +1,85 @@
+                  .medium-6.column
+                    .primary.callout.detail-wrapper.no-margin-bottom.no-padding-bottom
+                      .detail-edit
+                        = f.input :amount, as: :currency, input_html: { value: sprintf("%.2f", @client_data_instance.amount) }
```
we have this method in `app/helpers/value_helper.rb` (`property_to_s`) that returns decimal values as currency -- is that what you want here? This method is also used in `property_to_s` referenced above, so if we used that on all client data attrs this would show up as currency

jessieay replied:
seems this is a dupe of the `amount` field in `app/views/proposals/detail/fields/_amount.html.haml` ?

jessieay commented on app/views/proposals/detail/_request_details.html.haml:
```diff
@@ -0,0 +1,85 @@
+                              %h4
+                                %b.detail-value{id: 'amount-' + @client_data_instance.id.to_s }
+                                  = sprintf("%.2f", @client_data_instance.amount)
+            = submit_tag "Submit", hidden: true;
```
no need for semi-colon :smile:

jessieay commented on app/views/proposals/detail/_request_details.html.haml:
```diff
@@ -0,0 +1,85 @@
+                    .detail-wrapper
+                      %label.detail-element CL number
+                      .detail-edit
+                        = f.input :cl_number, label: false
```
we have this in `app/views/proposals/detail/fields/_cl_number.html.haml`...do we need both?

jessieay commented on app/views/proposals/detail/_request_details.html.haml:
```diff
@@ -0,0 +1,85 @@
+              = modify_client_button(@proposal, "<span>#{@mode == "edit" ? "View" : "Modify"}</span>".html_safe, "button secondary edit-icon fr large request-detail-edit-button")
+        .content-content.column#view-request-details
+          = simple_form_for @client_data_instance, html: { multipart: true }, remote: true do |f|
+            - excluded_values = ["direct_pay", "vendor", "building_number", "cl_number", "amount", "description"]
```
why are these excluded?

jessieay commented:
seems like there are some duplicate fields included in the form -- these also happen to all be labeled as ` `excluded_values` - what's that about?

jessieay commented:
I forked off your branch and added logic to get the dynamic fields to populate. 

I wasn’t sure what other attrs you wanted, so I just added the `partial` key to each hash. I got it working for both NCR and GSA. work is here https://github.com/18F/C2/compare/jy-request-details?expand=1

Hopefully this is enough to get you going on dynamically populating request fields. Can work together when you're back to add more stuff!

rememberlenny commented:
Thank you!!

--
Leonard Bogdonoff
949-322-8496 | @rememberlenny

> On Apr 15, 2016, at 8:38 PM, Jessie A. Young <notifications@github.com> wrote:
> 
> I forked off your branch and added logic to get the dynamic fields to populate.
> 
> I wasn’t sure what other attrs you wanted, so I just added the partial key to each hash. I got it working for both NCR and GSA. work is here https://github.com/18F/C2/compare/jy-request-details?expand=1
> 
> Hopefully this is enough to get you going on dynamically populating request fields. Can work together when you're back to add more stuff!
> 
> —
> You are receiving this because you were assigned.
> Reply to this email directly or view it on GitHub
>

pkarman commented:
looks like a whitespace cleanup needed for codeclimate.

rememberlenny commented:
@pkarman Im going to close this and replace with https://github.com/18F/C2/pull/1185
